src/main/java/executor/service/service/stepexecutors/selenium/SeleniumStepExecutionFactory.java

```diff
@@ -0,0 +1,33 @@
+package executor.service.service.stepexecutors.selenium;
+
+import executor.service.service.stepexecutors.StepExecution;
+import executor.service.service.stepexecutors.StepExecutionFabric;
+
+public class SeleniumStepExecutionFactory implements StepExecutionFabric {
+
+  private final StepExecutionClickCss css;
+  private final StepExecutionSleep sleep;
+  private final StepExecutionClickXpath xpath;
+
+  public SeleniumStepExecutionFactory(StepExecutionClickCss css, StepExecutionSleep sleep,
+      StepExecutionClickXpath xpath) {
+    this.css = css;
+    this.sleep = sleep;
+    this.xpath = xpath;
+  }
```

Comment:
I suppose switch/case or factory method would be enough. In case of Spring it can be set as follows:
`public StepExecution getStepExecution(Step step) {
        return beanFactory.getBean(step.getAction(),  StepExecution.class);
    }`

Reply:
There is FactoryForDI,  I think in this case this way is better.  Another case create stateless abstract class, which would return interfaceces

Reply:
I think should to write in style for the whole project, if class has foreign dependcies better inject there in constructor, another cases would have hidden dependcy.

If we will use abstract class it's better Kostya to rewrite, the task write DI factory is his.

Reply:
[Sokol3783] - I agree with you